client/rpcclient/service.go

```diff
@@ -30,13 +32,26 @@ func NewRPCClient(ctx context.Context, cfg *Config) *Service {
 		ctx:      ctx,
 		cancel:   cancel,
 		endpoint: cfg.Endpoint,
+		withCert: cfg.CertFlag,
 	}
 }
 
 // Start the grpc connection.
 func (s *Service) Start() {
 	log.Info("Starting service")
-	conn, err := grpc.Dial(s.endpoint, grpc.WithInsecure())
+	var server grpc.DialOption
+	if s.withCert != "" {
```

Comment:
use nil?

Reply:
This is the same problem as the previous, s.withCert is a string so it cannot be compared with nil.

Reply:
no worries, I'm looking if there's a better way to pass context from node level to service level